fix: Update migration guide and changelog for `executionDidEnd` (#5765)
apollographql/apollo-server: CHANGELOG.md

  - In a future release, `willResolveField` might become "sometimes-`async`" by returning a `ValueOrPromise`.
- Apollo Server now always fires the `willSendResponse` plugin lifecycle event after firing `didEncounterError`.
  - In certain error cases (mostly related to automated persisted queries), Apollo Server 2 skips firing `willSendResponse`.

> - Apollo Server no longer calls the "end hook" function returned by an `executionDidStart` event. The `executionDidStart` event should now return an object with an async function property called `executionDidEnd`.

Reviewer: ```suggestion
- The `executionDidStart` event can no longer return a function as an "end hook". The "end hook" for this event now must be provided as an async function property called `executionDidEnd` on an object.
```
maybe?
